travvy/src/components/SignUp.vue

```diff
@@ -45,42 +46,87 @@
                 email: '',
                 name: '',
                 password: '',
-                error: null,
+                errors: '',
             }
         }, 
         methods:{
+    
+            // this method takes in the user information and passes it to the authentication page 
+            // ; checkForm(), navigateTo({name:'Onboarding'});
             async signup(){
                 console.log(this.name)
                 console.log(this.email)
                 console.log(this.password)
+                console.log(this.errors)
+                
                 
                 try {
-                const response = await AuthenticationService.signup({
-                    "email": this.email,
-                    "name": this.name,
-                    "password": this.password,
-                })
-                console.log(response);
-                } catch (error) {
-                    console.log(error)
+                  const response = await AuthenticationService.signup({
+                    // this info inputted in the form is what is passed to the next page
+                      "email": this.email,
+                      "name": this.name,
+                      "password": this.password,
+                  })
+                  console.log(response)
+                  // this will add the user to the database and take them to the next page if the information is all inputted correctly
+                  if (response && this.checkForm(response)===true){
+                    this.$router.push("Onboarding")
+                  }else{
+                  //if the login is not valid, do nothing and tell the user it was wrong
+                    alert("The sign up is not valid. If you have no errors to correct - try a different email address as the one you used may already be in use.")
+                  }
+            
+              }catch (error) {
+                    alert(error)
+                   
                 }
             }, 
+            // this method checks for errors in the user input and will return the errors if any
+            checkForm: function(x){
+              // an array with the current errors and what will be returned to the user
+              this.errors = [];
+
+              // if a name is not entered, the site will tell the user a name is required
+              if(!this.name){
+                  this.errors.push("Name required");
```

Comment:
in future sprints we can make this more descriptive, users may be confused. But works perfectly for now.